msbuild/Xamarin.MacDev.Tasks.Core/Tasks/MetalTaskBase.cs

```diff
@@ -48,8 +51,22 @@ public abstract class MetalTaskBase : ToolTask
 			get;
 		}
 
-		protected abstract string OperatingSystem {
-			get;
+		protected virtual string OperatingSystem {
+			get {
+				switch (PlatformFrameworkHelper.GetFramework (TargetFrameworkIdentifier)) {
+				case PlatformFramework.WatchOS:
+					return "watchos";
+				case PlatformFramework.TVOS:
+					return "tvos";
+				case PlatformFramework.MacOS:
+					return "macosx";
+				case PlatformFramework.iOS:
+					return "ios";
+				default:
+					Log.LogError ($"Unknown target framework identifier: {TargetFrameworkIdentifier}.");
+					return string.Empty;
+				}
+			}
```

Comment:
That block is repeated 3 times (not DRY)
Can it be moved into a (new) base type ? or an helper method ?

Reply:
Actually it's not repeated 3 times, only 2 (and one version is different from the others). In any case I've moved the duplicated code into a helper method.

Reply:
Good point - but now you have to explain why this one is different ;-) and not shared
Curious minds **must** know :)

Reply:
Also, can we use c# 8? Switch expressions!

Reply:
@spouliot it's the mac value that's different (`osx` vs `macosx`). It was that way before my changes, and I didn't want to change anything, so I kept it that way.